src/main/java/graphql/schema/GraphQLTypeCodeVisitor.java

```diff
@@ -0,0 +1,51 @@
+package graphql.schema;
+
+import graphql.Internal;
+import graphql.util.TraversalControl;
+import graphql.util.TraverserContext;
+
+import static graphql.Assert.assertTrue;
+import static graphql.util.TraversalControl.CONTINUE;
+
+/**
+ * This ensure that all fields have data fetchers and that unions and interfaces have type resolvers
+ */
+@Internal
+class GraphQLTypeCodeVisitor extends GraphQLTypeVisitorStub {
+    private final GraphQLCodeRegistry.Builder codeRegistry;
+
+    GraphQLTypeCodeVisitor(GraphQLCodeRegistry.Builder codeRegistry) {
+        this.codeRegistry = codeRegistry;
+    }
+
+    @Override
+    public TraversalControl visitGraphQLFieldDefinition(GraphQLFieldDefinition node, TraverserContext<GraphQLType> context) {
+        GraphQLFieldsContainer parentContainerType = (GraphQLFieldsContainer) context.getParentContext().thisNode();
+        DataFetcher dataFetcher = node.getDataFetcher();
+        if (dataFetcher == null) {
+            dataFetcher = new PropertyDataFetcher<>(node.getName());
+        }
+        codeRegistry.dataFetcherIfAbsent(parentContainerType, node, dataFetcher);
+        return CONTINUE;
+    }
+
+    @Override
+    public TraversalControl visitGraphQLInterfaceType(GraphQLInterfaceType node, TraverserContext<GraphQLType> context) {
```

Comment:
why not using the same logic as for the DataFetcher and put the TypeResolver in the CodeRegistry?

Reply:
I do 

     codeRegistry.typeResolverIfAbsent(node, typeResolver);

This will put a type resolve against that type UNLESS it was previously seat directly by other code (eg direct put)